### PR TITLE
Update auth client configuration to use proper tls config

### PR DESCRIPTION
**Pull request by dmcgowan**
Currently the http clients used by auth use the default tls config.  The config needs to be updated to only support TLS1.0 and newer as well as respect registry insecure configuration.

### CODE REVIEW

**Comment by dmcgowan** [comment]
ping @jlhawn @diogomonica @nathanmccauley PTAL

**Comment by diogomonica** [comment]
@dmcgowan why not going to TLS1.2? Is there anything that doesn't support TLS1.2?

File: registry/auth.go [hunk 1]
```diff
@@ -70,10 +71,19 @@ func (auth *RequestAuthorization) getToken() (string, error) {
 		return auth.tokenCache, nil
 	}
 
+	tlsConfig := tls.Config{
+		MinVersion: tls.VersionTLS10,
```
Comment (jlhawn) [inline registry/auth.go]: @diogomonica suggested that we force TLS 1.2 as a minimum.

File: registry/auth.go [hunk 2]
```diff
@@ -362,10 +372,18 @@ func loginV1(authConfig *AuthConfig, registryEndpoint *Endpoint, factory *utils.
 func loginV2(authConfig *AuthConfig, registryEndpoint *Endpoint, factory *utils.HTTPRequestFactory) (string, error) {
 	log.Debugf("attempting v2 login to registry endpoint %s", registryEndpoint)
 
+	tlsConfig := tls.Config{
+		MinVersion: tls.VersionTLS10,
```
Comment (jlhawn) [inline registry/auth.go]: same here re:TLS1.2

File: registry/auth.go (hunk 1 above)
Comment (diogomonica) [inline registry/auth.go]: My recommendation is:

```go
config := tls.Config{
    MinVersion:               tls.VersionTLS10,
    PreferServerCipherSuites: true,
    CipherSuites: []uint16{
        tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
        tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
        tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
        tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
        tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
        tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
        tls.TLS_RSA_WITH_AES_128_CBC_SHA,
        tls.TLS_RSA_WITH_AES_256_CBC_SHA,
        tls.TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA},
}
```

**Comment by dmcgowan** [comment]
This change is to get the TLS configuration for auth in parity with the registry TLS configuration.  Any else is a separate change.

**Comment by diogomonica** [comment]
LGTM. I'll open the issues on Docker engine as a separate issue.

**Comment by dmp42** [comment]
LGTM pending janky happiness.

**Comment by jfrazelle** [comment]
no not until you fix pull of ubuntu:14.04 on master
cc @crosbymichael

**Comment by icecrime** [comment]
Ping @jfrazelle: pulling on master should be fixed

**Comment by jfrazelle** [comment]
LGTM

**Comment by icecrime** [comment]
After discussion with @diogomonica: PR will get updated to match his recommendation (TLS 1.2 + CipherSuites).

**Comment by dmp42** [comment]
@icecrime I would really prefer a separate PR for TLS cypher/version changes.
This PR here is about fixing a bug in the way we handle self-signed certificate for a very specific case - while upgrading cypher suites / TLS really is wider in extent.

**Comment by icecrime** [comment]
@dmp42 Works for me!

**Comment by icecrime** [comment]
A separate PR (not necessarily for 1.6) will followup: this is already an improvement worth merging.

**Pull request merged**